Operators, better error reporting, and excluding files

Zverik opened the pull request:
As it often happens, this pull request contains multiple fixes:

1. Support for operators in the binding generator. Fixes #252.
2. Paths listed in `analysis_options.yaml` are now skipped by the bindgen (usually there is generated code that produces analysis errors).
3. Skipping private fields and methods when generating bindings. Fixes #256.
4. Uri constructors now return Uri instances instead of strings.
5. Added three failing tests for operator methods, for the time we implement those.
6. Fixed `for` loops inside lists.
7. Added context for some errors, so that it's easier to find the failing statement or a type reference.
8. Fixed dart warnings, mainly from unnecessary labelling libraries.

Zverik commented:
Thanks for noticing! I've fixed the analyzer error, but not sure how to proceed with the deprecation of `RegExpMatch` inheritance.

ethanblake4 commented:
I guess the simplest solution right now would be to implement `Match` instead of `RegExpMatch`. Or, you could just remove the implements altogether and all the overridden methods... This style of bimodal wrapper that implements the superclass is something I'm slowly sort-of getting rid of, it seems incompatible with the Dart language future vision.

ethanblake4 commented:
Fwiw, I have reviewed the other changes and they are looking good.

I have a little bit of relucatance about adding additional dependencies, but I think `glob` is probably fine since it is from the Dart team. However, please be judicious about adding dependencies in the future, it should only be done if they are absolutely needed.

Zverik commented:
Fixed the `RegExpMatch` warning, and also fixed the `Uri` return types (just encountered that issue).

Regarding `glob` dependency, yes, I'm also wary of adding those, but I thought it's used by Dart and is pretty standard.

ethanblake4 commented:
Can you please run `dart format .` before this and other commits? Pub will get mad and it's a bit hard to read

Zverik commented:
Done :)
